src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -146,9 +155,19 @@ public boolean validCheckForObject(final AtlasObject object)
         // process all those edges
         explored.forEach(marked -> this.markAsFlagged(marked.getIdentifier()));
 
-        // Create the flag if and only if the empty flag value is not set to false
-        return emptyFlag ? Optional.empty()
-                : Optional.of(createFlag(explored, this.getLocalizedInstruction(0)));
+        // We want to filter out any Edges whose end Node has an amenity tag equal to one of the
+        // amenityValuesToExclude
+        final Set<AtlasObject> filteredExploredEdges = explored.stream()
+                .filter(edge -> validEdge(edge)).collect(Collectors.toSet());
+
+        if (!filteredExploredEdges.isEmpty())
+        {
+            // Create the flag if and only if the empty flag value is not set to false
+            return emptyFlag ? Optional.empty()
+                    : Optional
+                            .of(createFlag(filteredExploredEdges, this.getLocalizedInstruction(0)));
+        }
+        return Optional.empty();
```

Comment:
I don't think changes between lines 158 and 170 are needed. Since `validEdge` method is updated, now edges whose end node has an amenity tag will be ignored.

See line 109, which collects `outEdge` with `validEdge` filter.